main_app/views.py

```diff
@@ -0,0 +1,37 @@
+from django.shortcuts import render
+from django.views.generic import TemplateView
+
+from .igdb_api import IGDB
+
+
+class MainPageView(TemplateView):
+    template = 'main_page.html'
+
+    def __init__(self, **kwargs):
+        super(MainPageView, self).__init__(**kwargs)
+        self.client = IGDB(6)
+        self.__current_page = '1'
+        self.__search = ''
+        self.__platforms = ''
+        self.__genres = ''
+        self.__user_ratings = ''
+
+    def get_context_data(self, **kwargs):
+        super(MainPageView, self).get_context_data()
+        self.__current_page = self.request.GET.get('page', self.__current_page)
```

Comment:
just replace it with 
`current_page = self.request.GET.get('page', 1)`

You don't need to store this variable inside class. If several users will use your app, there will be problems due to how python instantiate classes.